From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 17 Sep 2025 19:06:55 +0200
Subject: gdm/util: Do not run dispose on user verifier

This is not a good habit anywhere and especially in JS code, I assume
the rationale for this was to ensure that cached instances of the proxy
were cleared, but this did not work properly (as it happened on
finalize), and it would still require [1].

So let's just avoid this and assume that the getters just return what we
expect.

[1] https://gitlab.gnome.org/GNOME/gdm/-/merge_requests/321

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3885>
(cherry picked from commit 7029cf41f1265a50009c237045aa36d73be1165b)
Origin: upstream, 49.4, commit:00df421871ceb5d2f380408447d25a1bd2e632d2
---
 js/gdm/util.js | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/js/gdm/util.js b/js/gdm/util.js
index f75c581..da05ecf 100644
--- a/js/gdm/util.js
+++ b/js/gdm/util.js
@@ -201,12 +201,8 @@ export class ShellUserVerifier extends Signals.EventEmitter {
         if (this._userVerifier) {
             this._disconnectSignals();
             this._userVerifier.get_connection().disconnectObject(this);
-            this._userVerifier.run_dispose();
             this._userVerifier = null;
-            if (this._userVerifierChoiceList) {
-                this._userVerifierChoiceList.run_dispose();
-                this._userVerifierChoiceList = null;
-            }
+            this._userVerifierChoiceList = null;
         }
     }
 
